docs/diagnostic-logs.md

```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
+
+By checking the log in server and service side, you can easily find out whether the message is sent from server, arrives at SignalR service, and leaves from SignalR service. Basically, by checking if the *received* and *sent* message are matched or not based on message tracing Id, you can tell whether the message loss issue is in server or SignalR service in this direction. For more information, see the [details](#message-flow-detail-for-path3) below.
+
+For **collect partially** collecting behavior:
+
+Once you mark the client as diagnostic client, SignalR service will trace messages in both directions. 
+
+By checking the log in server and service side, you can easily find out whether the message is pass the server or SignalR service successfully. Basically, by checking if the *received* and *sent* message are matched or not based on message tracing Id, you can tell whether the message loss issue is in server or SignalR service. For more information, see the details below.
+
+**Details of the message flow**
+
+For the direction **from client to server via SignalR service**, SignalR service will **only** consider the invocation that is originated from diagnostic client, that is, the message generated directly in diagnostic client, or service message generated due to the invocation of diagnostic client indirectly. 
+
+The tracing ID will be generated in SignalR service once the message arrives at SignalR service in **Path 1**. SignalR service will generate a log `Received a message <MessageTracingId> from client connection <ConnectionId>.` for each message in diagnostic client. Once the message leaves from the SignalR to server, SignalR service will generate a log `Sent a message <MessageTracingId> to server connection <ConnectionId> successfully.` If you see these 2 logs, you can be sure that the message passes through SignalR service successfully. 
+
+> Due to the limitation of ASP.NET Core SignalR, the message comes from client doesn't contains any message level ID. But ASP.NET SignalR generate *invocation ID* for each message, you can use it to map with the tracing ID.
+
+Then the message carries the tracing ID Server in **Path 2**. Server will generate a log `Received message <messagetracingId> from client connection <connectionId>` once the message arrives.
+
+<span id="message-flow-detail-for-path3"></span>
+Once the message invokes the hub method in server, a new service message will be generated with a *new tracing ID*. Once the service message is generated, server will generate a log in template `Start to broadcast/send message <MessageTracingId> ...`, the actual log will be based on your scenario. Then the message will be delivered to SignalR service in **Path 3**, once the service message leaves from server, a log called `Succeeded to send message <MessageTracingId>` will be generated. 
+
+> Due to the limitation of SignalR, message tracing ID can't be passed through SignalR hub. 
```

Comment:
what does this mean?

Reply:
I want to explain the ID of message from client cannot map to the ID of service message (service message will be sent to ASRS).
How about I change to:
> The tracing ID of message from client cannot map to the tracing ID of service message to be sent to SignalR service.

Reply:
This sounds better.